### PR TITLE
FSE - force private by default if in test-fse flow

### PR DESCRIPTION
**Pull request by glendaviesnz**
#### Changes proposed in this Pull Request

Force the privateByDefault for users on `test-fse` flow in order to bypass 90/10 AB test split.

#### Testing instructions

* Apply the patch to your FSE test environment
* Add some debugging to https://github.com/Automattic/wp-calypso/blob/master/client/lib/signup/private-by-default.ts#L19 to check that it bypasses abtest() on `test-fse` flow and returns true - site should always be created as private. The normal abtest() path should be followed when not on test-fse flow
* to test FSE flow start with https://dev-env/start/test-fse

### CODE REVIEW

File: client/lib/signup/private-by-default.ts [hunk 1]
```diff
@@ -8,15 +8,21 @@ import debugFactory from 'debug';
...
 const debug = debugFactory( 'calypso:signup:private-by-default' );
 
 /**
  * Should the site be private by default
  * @param siteType The selected site type / segment. Corresponds with the `slug` in ./site-type.js
+ * @param state The current client state
  * @returns `true` for private by default & `false` for not
  */
-export function shouldBePrivateByDefault( siteType: string = '' ): boolean {
+export function shouldBePrivateByDefault( siteType: string = '', state ): boolean {
```
Comment (glendaviesnz) [inline client/lib/signup/private-by-default.ts]: as the state object isn't actually used in these methods,  I didn't bother typing it. Could have added 
```
interface State {
	signup: Signup;
}

interface Signup {
	flow: Flow;
}

interface Flow {
	currentFlowName: string;
}
```
to indicate which slice of the state we were interested in, but seemed like overkill.

Comment (gwwar) [reply client/lib/signup/private-by-default.ts]: Generally we reserve passing state to full selectors in the `state/selectors` folder, with state being the first object passed. Maybe drop the changes in the file, and we can move the check to the caller in step-action.js. There's only one usage?

Comment (gwwar) [reply client/lib/signup/private-by-default.ts]: If we want the logic in one util function, we can alternatively pass through an optional flag like, `isFullSiteEditingEnabled`

File: client/lib/signup/step-actions.js [hunk 2]
```diff
@@ -163,7 +163,7 @@ export function createSiteWithCart( callback, dependencies, stepData, reduxStore
 				title: siteTitle,
 			},
 		},
-		public: getNewSitePublicSetting( siteType ),
+		public: getNewSitePublicSetting( siteType, state ),
```
Comment (gwwar) [inline client/lib/signup/step-actions.js]: So maybe:

```jsx
const isFullSiteEditingActivated = getCurrentFlowName( state ) === 'test-fse';
const isPublic = getNewSitePublicSetting( siteType ) && isFullSiteEditingActivated;
```

Another way that might be cleaner is forcing new users into the selected private by default test variant, when we enter the start/test-fse step. If we did this, would this ruin any testing results @jblz ?

Screen Shot 2019-08-13 at 12 03 20 PM

File: client/lib/signup/private-by-default.ts (hunk 1 above)
Comment (jblz) [reply client/lib/signup/private-by-default.ts]: It's probably best to go ahead and transition these two functions (`getNewSitePublicSetting` & `shouldBePrivateByDefault` into selectors. After all, the `siteType` param is directly from `getSiteType` here:
https://github.com/Automattic/wp-calypso/blob/a154341e3e4369952e2fadde123b35989fa052ed/client/lib/signup/step-actions.js#L146

File: client/lib/signup/step-actions.js (hunk 2 above)
Comment (jblz) [reply client/lib/signup/step-actions.js]: I'd probably just contain it to the flow as has been done here, so existing users maintain their grouping when traversing the main flow.

File: client/lib/signup/private-by-default.ts (hunk 1 above)
Comment (gwwar) [reply client/lib/signup/private-by-default.ts]: Gotcha @jblz that sounds better :D. 

@glendaviesnz let's try moving these two to the global selector folder then in `client/state/selectors` note that we'd want one selector per file, and global naming, where the function name makes sense without additional context.

Comment (glendaviesnz) [reply client/lib/signup/private-by-default.ts]: @gwwar & @jblz - have split into the two global selectors as suggested. It seems like at this stage that the shouldNewSiteBePrivateByDefault selector might be superfluous as nothing actually calls it - it appears that the original method was only exported for unit test purposes. 

Another option would have been to just have this as a private method of the getNewSitePublicSetting selector instead - but maybe there is a future use of this selector in its own right that makes it worth separating out ? Happy to refactor into a single selector if that makes more sense.

**Comment by jblz** [comment]
Not really blocking, but I left some changes I think we should make in #35375.  Feel free to either merge into here before merging this to master or rebase #35375 on master after.

**Comment by gwwar** [comment]
Thanks @jblz we're going to land this one first, and then help out in https://github.com/Automattic/wp-calypso/pull/35375 as a follow up.

**Pull request merged**